### ISSUE TITLE
`embed-resources: true` fails with `-o -`

### ISSUE THREAD

**Issue by kdheepak** [quarto-dev/quarto-cli#11068]
### Bug description

![Image](https://github.com/user-attachments/assets/da891e23-8916-4de5-a632-6cc5bdd0bc14)

I get this error after the contents of the html output are printed to the terminal.

### Steps to reproduce

Download this file:

https://github.com/kdheepak/kdheepak.github.io/blob/main/blog/the-egg-tower-puzzle/index.ipynb

Run `quarto render index.ipynb --output -`

### Quarto check output

Quarto 1.5.38
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK

**Comment by mcanouil** [comment]
Could you upgrade to a stable release?
You are using an old pre-release: https://github.com/quarto-dev/quarto-cli/releases/tag/v1.5.38

**Comment by mcanouil** [comment]
Note that I cannot reproduce at least on the development version:

```jupyter
{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"hello world\")"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
```

![Image](https://github.com/user-attachments/assets/d92e5fb8-e113-403c-83ac-8a647dc626d6)

**Comment by kdheepak** [comment]
Thanks for the quick response and for checking.

Actually, I'm realizing it only happens with `embed-resources: true`. 

```
format:
  html:
    embed-resources: true
```

Can you try adding the above to the same notebook?

**Comment by cscheid** [comment]
That latter .ipynb file appears to not be well-formed, or at least it's not doing here what you appear to be reporting:

```
$ quarto render issue-11068.ipynb --to html --execute
ERROR: TypeError: Cannot read properties of undefined (reading 'name')
```

**Comment by cscheid** [comment]
With a well-formed notebook that does the same thing, I still cannot repro:

```
{
 "cells": [
  {
   "cell_type": "raw",
   "id": "4ac0767a-170b-47cc-b84a-a8b7a4e43d9a",
   "metadata": {},
   "source": [
    "---\n",
    "format:\n",
    "  html:\n",
    "    embed-resources: true\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "10db3539-d318-4837-ac73-8eedbc09f586",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hello world\n"
     ]
    }
   ],
   "source": [
    "print(\"hello world\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
```

```
$ quarto render issue-11068-good.ipynb --to html --execute
Starting python3 kernel...Done

Executing 'issue-11068-good.ipynb'
  Cell 1/1: ''...Done

pandoc
  to: html
  output-file: issue-11068-good.html
  standalone: true
  embed-resources: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en

Output created: issue-11068-good.html
```

Note that the output has `embed-resources: true`.

**Comment by kdheepak** [comment]
hmm, I copy pasted your codeblock into my `demo.ipynb` and with the latest pre-release I'm still getting an error:

![Image](https://github.com/user-attachments/assets/933d45b7-9790-496a-a5ff-e49167456f1a)

![Image](https://github.com/user-attachments/assets/503f8e51-d825-43b7-ab11-40479c911c41)

**Comment by mcanouil** [comment]
Could you provide the full output of `quarto check`?

**Comment by cscheid** [comment]
Ah, sorry, this is on me.

**Comment by cscheid** [comment]
This is pretty simple - `embed-resources: true` is not compatible with writing to standard output, at least not right now.

**Comment by cscheid** [comment]
The bug is here: https://github.com/quarto-dev/quarto-cli/blame/main/src/command/render/output.ts#L211

I'm a little surprised by it, because it's been there for 4 years, and there's a comment a few lines above explaining that we shouldn't remove the file precisely because it might be necessary.

**Comment by cscheid** [comment]
Thanks for the report! This is fixed now on `main` and a new 1.6 prerelease build is coming out momentarily.

**Comment by kdheepak** [comment]
Thanks for the quick fix!
